### ISSUE TITLE
Some requests

### ISSUE THREAD

**Issue by JohnGriffiths** [srothmei/TVB-Pypeline#2]
Hey Simon. 

Could you please add to the repo somewhere:

1. A general description of the implemented + intended overall design: what goes where; what does what, etc.
2. A to-do list of some sort so I can see what needs doing and where my efforts would be most useful
3. A 'getting started' example

Cheers, 

j

**Comment by srothmei** [comment]
Hey John,

1.) The general description so far is that im currently just trying to translate the old shell-script-based workflow into nipype. 
What is already in are the mrtrix tracking module, the preprocessing steps and also the aggregation steps. What is missing (see attached workflow diagram or the TVB pipeline repo) is the processing of the functional data i.e. a workflow for that. The steps are listed in this specific file:
https://github.com/BrainModes/TVB-empirical-data-pipeline/blob/NSG/fmriFC.sh

The intended overall design i have in mind is that for each step in the current shell-script pipeline, we have separate workflows. Inside these sub-workflows, there are sub-sub-workflows wrapping the steps belonging to a single toolbox (i.e. mrtrix) such that the specific toolbox can easily be changed by just manipulating a single point in the workflow. 
This also requires that the input/output into the toolboxes is kept as generic as possible, e.g. by outputting Trackvis files out of the tracking module instead of for example MRTrix-speficic tsk files

2.) If you want to you could translate the fMRI part based on the aforementioned shell script. :-)

3.) You mean an example for running the pipeline? I intended to write the documentation as soon as the pipeline is running. 

![supfig1](https://cloud.githubusercontent.com/assets/7510769/11391315/e98e11fa-9351-11e5-8f36-2873432bc1cc.png)

**Comment by JohnGriffiths** [comment]
Actually if you don't mind I'd like to work towards getting the diffusion and T1 processing working smoothly before moving on to the fMRI. 

Looks like `TVB_pipeline.ipynb` is the entry point at present.  Is that the correct?

**Comment by srothmei** [comment]
Yes exactly. I also pushed my latest iteration of the code to the repo just this very second.
T1 and diffusion workflows are now defined, at least on the paper. Testing is still pending.

Two things that i also would like to do are:
1.) Make the input more dynamic. By this i mean that for example currently, the users raw-data have to match a certain folder structure. It would be good if the Pipeline recognizes this structure automatically. I think this could be achieved using the Data-Grabber/Finder tools from Nipype
2.) Rewrite the mask generation script such that its not hard-coded with Desikan-Killaney anymore. Currently, some definitions/loops inside there are still tailored to work with the 68 Region DK atlas. This has to be changed of course. Best thing would the to just input the parcellation of GM areas in T1-space and the script will do all the rest...

**Comment by JohnGriffiths** [comment]
Dude trying to get in touch outside of github issues but the e-mail I have for you is bouncing. 

So I'll just message here. 

I think we need an additional non-issue chat-type communication line. Suggest google chat or possibly gitter. Don't know much about gitter but have made a test for my repo fork; you should have got the invite. If that works ok we should probably scratch it though and make one for this repo instead.
